rfcs/rfc-colin-states-mapping-to-lear.md

```diff
@@ -0,0 +1,240 @@
+                'limitedRestoration': ['liquidation'], 
+            },
+            'putBackOn': ['liquidation']
+            ...
+        }
+    }        
+    
+}
+```
+
+_Allowable state transitions matrix in COLIN for restoration and put back on filings_
+![colin_allowable_filings.png](rfc-colin-states-mapping-to-lear/colin_allowable_filings.png)
+
+Filing validation logic will be updated to include logic to ensure that only valid business state transitions are allowed. 
+
+All filings resulting in business state transitions will need to populate the businesses.state_filing_id field.  This is used to determine if future business state transitions are permitted.
+
+Filings.meta_data field for any filings where a business state transitions occurs will include from/to state.
+
+API consumers such as the FE with a need to determine which filing transitioned the business to its current state will retrieve the business state filing provided by the business response of the GET business endpoint.  For example, this can be used to determine if an ACTIVE business is currently in _limited restoration_.
+
+Flow statuses such as `good standing`, `liquidation`, `involuntary dissolution` will make use of the existing warnings structure.  This will provide a way to track where a business is within a given flow.
+
+_Examples of how this might look.  Note these are only examples and the actual business rules will likely be different._
+
+##### Not In Good Standing
+
+``` json
+{
+    "business": {
+        "legalName": "XYZ TEST CORP.",
+        "legalType": "BC",
+        "state": "ACTIVE",
+        "goodStanding": false,
+        "warnings": [
+            {
+                "code": "ONE_ANNUAL_REPORT_NOT_FILED",
+                "message": "Latest annual report missing.",
+                "warningType": "GOOD_STANDING"
+            }       
+        ],
+        ...
+    }
+}
+
+```
+
+##### Involuntary Dissolution
+Strictly as an example, the `MULTIPLE_ANNUAL_REPORTS_NOT_FILED` warning could be used an indicator that the involuntary dissolution process needs to be initiated.
+
+``` json
+{
+    "business": {
+        "legalName": "XYZ TEST CORP.",
+        "legalType": "BC",
+        "state": "ACTIVE",
+        "goodStanding": false,
+        "warnings": [
+            {
+                "code": "MULTIPLE_ANNUAL_REPORTS_NOT_FILED",
+                "message": "Multiple annual reports not filed.",
+                "warningType": "GOOD_STANDING"
+            },
+            {
+                "code": "INVOLUNTARY_DISSOLUTION_STEP_1",
+                "message": "Involuntary dissolution process has been initiated.",
+                "warningType": "INVOLUNTARY_DISSOLUTION_FLOW"
+            }            
+        ],
+        ...
+    }
+}
+```
+
+#####  Liquidation
+
+``` json
+{
+    "business": {
+        "legalName": "XYZ TEST CORP.",
+        "legalType": "BC",
+        "state": "ACTIVE",
+        "warnings": [
```

Comment:
have we had much feedback on the _warnings_ block.
I'm digging our move to that as we expand on it here it seems to be meeting our needs, and was worth the migrations.